src/PrestaShopBundle/Form/Admin/Product/ProductShipping.php

```diff
@@ -68,7 +68,12 @@ public function __construct($translator, $legacyContext, $warehouseDataProvider,
         );
         $this->carriersChoices = [];
         foreach ($carriers as $carrier) {
-            $this->carriersChoices[$carrier['name'] . ' (' . $carrier['delay'] . ')'] = $carrier['id_reference'];
+            $choiceId = $carrier['id_carrier'] . ' - ' . $carrier['name'];
```

Comment:
Duplicate code :thinking: 
Why not using CarrierByReferenceChoiceProvider.php only?

Reply:
It's how the infamous Product page has been built 😭

Reply:
:nauseated_face: 
Maybe time to refactor :/

Reply:
Soon 😄